src/charm.py

```diff
@@ -28,7 +30,7 @@ class MySQLCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
-        self._stored.set_default(mysql_setup={})
+        self._stored.set_default(mysql_setup={"MYSQL_ROOT_PASSWORD": False})
```

Comment:
I guess this means you cannot make persistent changes in config data from charm code?
If you could rely on some other construct, such as `env_config` to store the password it would probably be better than `_stored` (which is duplicated across units?)

Reply:
I am using the `_stored` variable just to store this value between executions of the same unit. This value finally is assigned in `env_config` to be used in the pebble layer